src/Components/Authentication/Signup/signup.jsx

```diff
@@ -1,121 +1,201 @@
+    email: "",
+    password: "",
+    confirmPassword: "",
+  };
+  const initialErrorData = {
+    firstNameError: "",
+    lastNameError: "",
+    emailError: "",
+    passwordError: "",
+    confirmPasswordError: "",
+    serverError: "",
+  };
   const [signupForm, setSignupForm] = useState(initialFormData);
-  const [errorData, setErrorData] = useState(initialErrorData)
-  const { firstNameError, lastNameError, emailError, passwordError, confirmPasswordError, serverError } = errorData;
+  const [errorData, setErrorData] = useState(initialErrorData);
+  const {
+    firstNameError,
+    lastNameError,
+    emailError,
+    passwordError,
+    confirmPasswordError,
+    serverError,
+  } = errorData;
   const navigate = useNavigate();
   const { setLogin } = useAuthenticationContext();
 
   function formHandler(e) {
     const { name, value } = e.target;
     if (name === "confirmPassword" && signupForm.password !== value)
-      setErrorData(err => ({ ...err, confirmPasswordError: "Password do not match " }))
+      setErrorData((err) => ({
+        ...err,
+        confirmPasswordError: "Password do not match ",
+      }));
     if (name === "password" && signupForm.confirmPassword == value)
-      setErrorData(err => ({ ...err, confirmPasswordError: "" }))
-
+      setErrorData((err) => ({ ...err, confirmPasswordError: "" }));
     else {
-      setSignupForm(formValues => ({ ...formValues, [name]: value }))
-      setError(`${name}Error`, "")
+      setSignupForm((formValues) => ({ ...formValues, [name]: value }));
+      setError(`${name}Error`, "");
     }
   }
 
   function setError(name, value) {
-    setErrorData(err => ({ ...err, [name]: value }))
+    setErrorData((err) => ({ ...err, [name]: value }));
   }
 
   function signupHandler(e) {
     e.preventDefault();
-    const validData = signupValidationHandler(setError, setErrorData, signupForm);
+    const validData = signupValidationHandler(
+      setError,
+      setErrorData,
+      signupForm
+    );
     const { firstName, lastName, email, password } = signupForm;
 
     if (validData) {
       (async () => {
         try {
-          const { data: { encodedToken } } = await axios.post('api/auth/signup', { firstName, lastName, email, password, });
+          const {
+            data: { encodedToken },
+          } = await axios.post("api/auth/signup", {
+            firstName,
+            lastName,
+            email,
+            password,
+          });
           if (encodedToken) {
             localStorage.setItem("token", encodedToken);
             setLogin(true);
             setSignupForm(initialFormData);
             navigate("/");
           }
-        }
-        catch (err) {
-          err.response.status === 422 ? setError("serverError", "Email already exists") : setError("serverError", "Can't connect to server ! Try again later")
+        } catch (err) {
+          err.response.status === 422
+            ? setError("serverError", "Email already exists")
```

Comment:
422 is not a server error. It's a client error. You may want to show it differently.

Reply:
Just a suggestion. Not a blocker to merge the PR.

Reply:
Yes ,I know that, but it's a bug in mockbee, you will get 422 in catch block only if email exists